### ISSUE TITLE
Consider making the API async

### ISSUE THREAD

**Issue by uazo** [explainers-by-googlers/partitioned-popins#11]
The proposal uses `window.open` which is synchronous.
This would not easily allow the development of browsers that require authorisation from the user before opening the popin.
Would it be possible to rethink the api call?

**Comment by arichiv** [comment]
Could you elaborate on the flow you're describing? I'm not against using something besides window.open, it just seemed like the best current match.

**Comment by uazo** [comment]
What I propose is to evaluate a call like
```
await window.openPopip(...);
```
therefore return a promise.

the advantages I believe are:
- using an async API is more modern, [w3c](https://www.w3.org/2001/tag/doc/promises-guide) specifies it best
- you can eliminate support for the [detection](https://github.com/explainers-by-googlers/partitioned-popins?tab=readme-ov-file#detection) you are proposing
- potentially not introduce regressions in browsers that do not support popins
- is more easily exploited (development-side) for possible asynchronous user request management

**Comment by johannhof** [comment]
I think conceptually I like the idea of getting a Promise from the opened pop-in, returning a result from the user interaction that happened in the pop-in seems like a common pattern and making that more modern / ergonomic feels worthwhile.

With that said, maybe there are advantages to utilizing the existing API that aren't apparent at first glance. Probably worth investigating!

**Comment by uazo** [comment]
interesting to note that suggestions are of no use...

https://issues.chromium.org/issues/340606651

so what is this repo for?

**Comment by arichiv** [comment]
I disagree that suggestions are of no use, but not all suggestions will be taken and those taken may not be adopted on the timeline requested.

This project is still just being prototyped, and has not even entered the origin trial phase (let alone contemplating shipping). Feedback not addressed so far may be addressed in the future.

Also, the more feedback on the same point we get from multiple developers/vendors the more prioritized that piece of feedback is likely to be.
